include/operations/blas_operators.hpp

```diff
@@ -55,78 +55,31 @@ namespace blas {
+                                                               \
+    template <typename R>                                      \
+    static typename R::value_type init(const R) {              \
+      return constant<typename R::value_type, initial>::value; \
+    }                                                          \
   };
 
-/* strip_asp.
- * When using ComputeCpp CE, the Device Compiler uses Address Spaces
- * to deal with the different global memories.
- * However, this causes problem with std type traits, which see the
- * types with address space qualifiers as different from the C++
- * standard types.
- *
- * This is strip_asp function servers as a workaround that removes
- * the address space for various types.
- */
-template <typename TypeWithAddressSpace>
-struct strip_asp {
-  typedef TypeWithAddressSpace type;
-};
-
-#if defined(__SYCL_DEVICE_ONLY__) && defined(__COMPUTECPP__)
-#define GENERATE_STRIP_ASP(ENTRY_TYPE)                             \
-  template <>                                                      \
-  struct strip_asp<__attribute__((address_space(1))) ENTRY_TYPE> { \
-    typedef ENTRY_TYPE type;                                       \
-  };                                                               \
-                                                                   \
-  template <>                                                      \
-  struct strip_asp<__attribute__((address_space(2))) ENTRY_TYPE> { \
-    typedef ENTRY_TYPE type;                                       \
-  };                                                               \
-                                                                   \
-  template <>                                                      \
-  struct strip_asp<__attribute__((address_space(3))) ENTRY_TYPE> { \
-    typedef ENTRY_TYPE type;                                       \
-  };
-
-GENERATE_STRIP_ASP(IndVal<double>)
-GENERATE_STRIP_ASP(IndVal<float>)
-GENERATE_STRIP_ASP(double)
-GENERATE_STRIP_ASP(float)
-#endif  // __SYCL_DEVICE_ONLY__  && __COMPUTECPP__
-
-/**
- * syclblas_abs.
- *
- * SYCL 1.2 defines different functions for abs for floating point
- * and integer numbers, following the OpenCL convention.
- * To choose the appropriate one we use this template specialization
- * that is enabled for floating point to use fabs, and abs for everything else.
+/** wang random generator
  */
-struct syclblas_abs {
-  template <typename Type>
-  static Type eval(const Type &val,
-                   typename std::enable_if<!std::is_floating_point<
-                       typename strip_asp<Type>::type>::value>::type * = 0) {
-    return cl::sycl::abs(val);
-  }
-
-  template <typename Type>
-  static Type eval(const Type &val,
-                   typename std::enable_if<std::is_floating_point<
-                       typename strip_asp<Type>::type>::value>::type * = 0) {
-    return cl::sycl::fabs(val);
+struct wang {
```

Comment:
How do you compute the abs function in SYCLBLAS?
Morover, I don't know the meaning of wang